### PR TITLE
Allow setting HTTP request method to make caching possible

### PR DESCRIPTION
**Pull request by cvhoang**

### CODE REVIEW

File: sunspot/lib/sunspot/search/abstract_search.rb [hunk 1]
```diff
@@ -46,7 +46,12 @@ def initialize(connection, setup, query, configuration) #:nodoc:
       def execute
         reset
         params = @query.to_params
-        @solr_result = @connection.post "#{request_handler}", :data => params
+
+        if @configuration.solr.http_method.to_s == 'post'
+          @solr_result = @connection.post "#{request_handler}", :data => params
```
Comment (jobseekerhound) [inline sunspot/lib/sunspot/search/abstract_search.rb]: Use the new Ruby 1.9 hash syntax.

File: sunspot/lib/sunspot/search/abstract_search.rb [hunk 2]
```diff
@@ -46,7 +46,12 @@ def initialize(connection, setup, query, configuration) #:nodoc:
...
+        else
+          @solr_result = @connection.get "#{request_handler}", :params => params
```
Comment (jobseekerhound) [inline sunspot/lib/sunspot/search/abstract_search.rb]: Use the new Ruby 1.9 hash syntax.

**Comment by ggasser** [comment]
Don't we need some testing for this?

**Comment by cvhoang** [comment]
Oh I hand-merged the pull request from sunspot.

**Pull request merged**